### PR TITLE
Fix panic when telemetry metrics are disabled

### LINKED ISSUE #5641: Collector panics if setting metrics level to 'none'
**Describe the bug**

The OpenTelemetry Collector will panic if you set `telemetry::metrics::level` to `none`.

**Steps to reproduce**

Run a Collector with the following configuration:

```
receivers:
  otlp:
    protocols:
      grpc:

exporters:
  logging:

service:
  telemetry:
    metrics:
      level: none
  pipelines:
    metrics:
      receivers: [otlp]
      exporters: [logging]
```

**What did you expect to see?**

The Collector starts running as usual.

**What did you see instead?**

The Collector panics:

```
2022-07-07T09:01:24.200Z	info	service/telemetry.go:95	Skipping telemetry setup.	{"address": ":8888", "level": "none"}
2022-07-07T09:01:24.200Z	info	pipelines/pipelines.go:341	Component is under development.	{"kind": "exporter", "data_type": "metrics", "name": "logging", "stability": "in development"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb0e749]

goroutine 1 [running]:
go.opencensus.io/metric.(*Registry).initBaseMetric(0x0, 0xc000772ea0, {0x32dc476, 0xe}, {0xc000c1f2a0, 0x2, 0x3929788?})
	go.opencensus.io@v0.23.0/metric/registry.go:238 +0x69
go.opencensus.io/metric.(*Registry).AddFloat64DerivedCumulative(0x3929788?, {0x32dc476, 0xe}, {0xc000c1f2a0, 0x2, 0x2})
	go.opencensus.io@v0.23.0/metric/registry.go:222 +0x7e
go.opentelemetry.io/collector/service/internal/telemetry.RegisterProcessMetrics(0x392ae10?, 0x0)
	go.opentelemetry.io/collector@v0.55.0/service/internal/telemetry/process_telemetry.go:61 +0x225
go.opentelemetry.io/collector/service.newService(0xc00091f7d0)
	go.opentelemetry.io/collector@v0.55.0/service/service.go:101 +0x9fa
go.opentelemetry.io/collector/service.(*Collector).setupConfigurationComponents(0xc000c1fae8, {0x3929788, 0xc000052018})
	go.opentelemetry.io/collector@v0.55.0/service/collector.go:184 +0x173
go.opentelemetry.io/collector/service.(*Collector).Run(0xc000c1fae8, {0x3929788, 0xc000052018})
	go.opentelemetry.io/collector@v0.55.0/service/collector.go:210 +0x45
go.opentelemetry.io/collector/service.NewCommand.func1(0xc00075e780, {0x32cdf01?, 0x2?, 0x2?})
	go.opentelemetry.io/collector@v0.55.0/service/command.go:50 +0x499
github.com/spf13/cobra.(*Command).execute(0xc00075e780, {0xc00004c1f0, 0x2, 0x2})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc00075e780)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
main.runInteractive({{0xc0007700c0, 0xc0007705d0, 0xc000770420, 0xc00074bc20}, {{0x32cd7ad, 0x7}, {0x32f3642, 0x17}, {0x32ca411, 0x6}}, ...})
	github.com/open-telemetry/opentelemetry-collector-releases/core/main.go:32 +0x5d
main.run(...)
	github.com/open-telemetry/opentelemetry-collector-releases/core/main_others.go:11
main.main()
	github.com/open-telemetry/opentelemetry-collector-releases/core/main.go:25 +0x1d8
```

**What version did you use?**

This started on v0.55.0, I believe the culprit is #5571.

**What config did you use?**

See above

**Environment**

Not relevant

**Additional context**

The issue is that registering process metrics is done irrespective of what the metrics level is: https://github.com/open-telemetry/opentelemetry-collector/blob/a1e2ad0543d89ab5e182daeb57b9243b63a2db11/service/service.go#L100-L103

A workaround is to set the level above none. If there is a port clash, you can modify the telemetry address to point to another port.

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** Fix panic when `telemetry::metrics::level` is set to `none`.

**Link to tracking Issue:** Fixes #5641

**Testing:** Add unit test for avoiding regressions

### CODE REVIEW

**Comment by mx-psi** [comment]
As seen in [the CI logs](https://github.com/open-telemetry/opentelemetry-collector/runs/7230412394?check_suite_focus=true#step:6:4342) the unit test reproduces the crash

File: service/collector_test.go [hunk 1]
```diff
@@ -373,6 +373,29 @@ func TestCollectorShutdownBeforeRun(t *testing.T) {
...
+	cfgProvider, err := NewConfigProvider(newDefaultConfigProviderSettings([]string{filepath.Join("testdata", "otelcol-nop.yaml")}))
+	require.NoError(t, err)
+
+	set := CollectorSettings{
+		BuildInfo:      component.NewDefaultBuildInfo(),
+		Factories:      factories,
+		ConfigProvider: cfgProvider,
+		telemetry:      newColTelemetry(featuregate.NewRegistry()),
+	}
+	col, err := New(set)
```
Comment (jpkrohling) [inline service/collector_test.go:389]: There's something odd -- have you swapped the test names by any chance?

Comment (jpkrohling) [reply service/collector_test.go:389]: I use the split view which should prevent this but I checked out the actual files in any case: the older version has `require.NotPanics(t, func() { col.Shutdown() })` in `TestCollectorShutdownBeforeRun`, which the new doesn't. This line does show up in `TestCollectorWithNoMetrics` though.

Comment (mx-psi) [reply service/collector_test.go]: You are right, thanks for catching that! Fixed

File: service/service.go [hunk 2]
```diff
@@ -97,9 +98,11 @@ func newService(set *settings) (*service, error) {
 		return nil, fmt.Errorf("cannot build pipelines: %w", err)
 	}
 
-	// The process telemetry initialization requires the ballast size, which is available after the extensions are initialized.
-	if err = telemetry.RegisterProcessMetrics(srv.telemetryInitializer.ocRegistry, getBallastSize(srv.host)); err != nil {
-		return nil, fmt.Errorf("failed to register process metrics: %w", err)
+	if set.Config.Telemetry.Metrics.Level != configtelemetry.LevelNone && set.Config.Telemetry.Metrics.Address != "" {
```
Comment (TylerHelmuth) [inline service/service.go:101]: Should we also have a test that covers the scenario of the  metric level being > None but the address not being set?  Its a pretty silly scenario but the if statement is checking for it.

**Review by TylerHelmuth — APPROVED** [review]
LGTM

File: service/host_test.go.bak [hunk 3]
```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply service/host_test.go.bak]: Yes, it shouldn't be here

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

File: service/service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mx-psi) [reply service/service.go:101]: #5655

[the remark this answers is not in the document]
Comment (codeboten) [reply service/service.go:101]: Thanks! Yeah I eventually tracked down the same check :D
